### PR TITLE
User views a list of all books

### PR DESCRIPTION
**Pull request by samnang**
https://trello.com/c/Yj3i4Q4I

Pair with @evan-007 

* We have to run `rails generate refills:import flashes` to pull flashes partial from refills gem

NOTE: We tried to add validation on book's title, but we should get shoulda-matcher working on model spec:
```ruby
RSpec.describe Book, :type => :model do
  it { is_expected.to validate_presence_of(:title) }
end
```

And here we got:

```
NoMethodError: undefined method `validate_presence_of'
```

### CODE REVIEW

**Comment by evan-007** [comment]
looks good to me, I had the same issue with rspec when i tried this locally, I'll dig into it tomorrow and see what I can find.

**Comment by evan-007** [comment]
I was able to fix that shoulda error by not using v3.1, but I'm still not sure what the issue is.

```
gem "shoulda-matchers", '< 3.0.0', require: false
```

**Comment by samnang** [comment]
@evan-007 if so, it's better to create an issue in their repo to see if somebody faced the same problem like us before.

**Comment by samnang** [comment]
I could dig down as well when I have times.

**Comment by evan-007** [comment]
I think the version of suspenders that I used just didn't setup shoulda-matchers 3.0 correctly:


https://robots.thoughtbot.com/shoulda-matchers-3-0

No longer need require: false + should add this config into `spec_helper`:

``` ruby
Shoulda::Matchers.configure do |config|
  config.integrate do |with|
    # Choose a test framework:
    with.test_framework :rspec
    with.test_framework :minitest
    with.test_framework :minitest_4
    with.test_framework :test_unit

    # Choose a library:
    with.library :active_record
    with.library :active_model
    with.library :action_controller
    # Or, choose all of the above:
    with.library :rails
  end
end
```

I'm in the office all day today, but I will try it tomorrow morning/tonight if no one else gets to it before me

**Comment by samnang** [comment]
@evan-007 When I dig down the source, I see there no any auto initialize hook to setup shoulda matcher to testing framework like RSpec, so we need to configure it. See this commit https://github.com/ajkamel/cover/commit/2909390953e94433a127ccdd7339ecf96b01be24 Or https://github.com/thoughtbot/shoulda-matchers#configuration

**Comment by samnang** [comment]
Fix the test when we merge this https://github.com/ajkamel/cover/pull/5

File: spec/models/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (evan-007) [inline spec/models/book_spec.rb]: How would you guys feel about removing this setting from hound? @samnang @ajkamel

Comment (ajkamel) [inline spec/models/book_spec.rb]: @evan-007 I'm indifferent, I usually use single as well.  It seems like they address it in this issue.  https://github.com/thoughtbot/hound/issues/977

Comment (evan-007) [inline spec/models/book_spec.rb]: i opened https://github.com/ajkamel/cover/pull/6 in case we want to change any of the settings, i prefer single.

**Comment by evan-007** [comment]
ah great, i'm glad you were able to figure it out. I merged #5 so feel free to merge this one in if you want.

File: spec/models/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+RSpec.describe Book, :type => :model do
```
Comment (ajkamel) [inline spec/models/book_spec.rb]: key: :value  

v. 

:key => :value

File: spec/models/book_spec.rb (hunk 1 above)
Comment (samnang) [inline spec/models/book_spec.rb]: Actually, I like single quote as well, but one of my team currently just stick with double quotes all the time. Either way should be fine with me.

File: spec/models/book_spec.rb (hunk 2 above)
Comment (samnang) [inline spec/models/book_spec.rb]: I love the new Ruby 1.9 hash syntax even it looks wired when both key and value are symbols.

**Pull request merged**